src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { memo, useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+export type Props = {
+  todo: Todo
+  deletedTodo: (value: number) => void
+  deletingTodoIds: number[]
+  onUpdatingTodo: (value: Todo) => void
+  updatingTodoIds: number[]
+};
+
+export const TodoItem: React.FC<Props> = memo(({
+  todo,
+  deletedTodo,
+  deletingTodoIds,
+  onUpdatingTodo,
+  updatingTodoIds,
+}) => {
+  const [todoTitleField, setTodoTitleField] = useState(todo.title);
+  const [isEditing, setIsEditing] = useState(true);
```

Comment:
I'm not sure but maybe this state should be `false` by default. Last todo on your demo-link is always editing. Even if it is not double-clicked. The same mistake when i add new todo - it is being edited. Please fix this

Reply:
so much thankfully, you were totally right with boolean value